cmd/virt-handler/virt-handler.go

```diff
@@ -306,14 +305,28 @@ func (app *virtHandlerApp) Run() {
 
 	stop := make(chan struct{})
 	defer close(stop)
-	var capabilities *api.Capabilities
+	var capabilities libvirtxml.Caps
 	var hostCpuModel string
+
+	hostCapsFile, err := os.ReadFile(filepath.Join(nodelabeller.NodeLabellerVolumePath, "capabilities.xml"))
+	if err != nil {
+		panic(err)
+	}
+
+	if err := capabilities.Unmarshal(string(hostCapsFile)); err != nil {
+		panic(err)
+	}
+
 	nodeLabellerrecorder := broadcaster.NewRecorder(scheme.Scheme, k8sv1.EventSource{Component: "node-labeller", Host: app.HostOverride})
-	nodeLabellerController, err := nodelabeller.NewNodeLabeller(app.clusterConfig, app.virtCli.CoreV1().Nodes(), app.HostOverride, nodeLabellerrecorder)
+	nodeLabellerController, err := nodelabeller.NewNodeLabeller(app.clusterConfig,
+		app.virtCli.CoreV1().Nodes(),
+		app.HostOverride,
+		nodeLabellerrecorder,
+		capabilities.Host.CPU.Counter,
+	)
```

Comment:
> I think it would be cleaner if a separate "unit" would be in charge of the inference (package / function) instead of doing so in the Run() function which is already pretty long and non-cohesive. Bear in mind that in the future more inference code will be added (e.g. for dom capabilities, host supported features, etc).

I agree, I wasn't sure whether to introduce it during this PR or not but we can do that.
My idea was for a new package under virt-handler (e.g. `NodeCapabilityManager`) and slowly migrate `cpu_plugin.go` logic into it WDYT?

> It's confusing that the mount path and even the init container's script is still called node-labeller. Perhaps we can rename all of these to more informative names. I think it's important since then it will be clearer of what your long-term plan is.

I agree, I'll add it in a separate commit.

> Further to the above, I think that nodelabeller.NodeLabellerVolumePath should now reside outside of node-labeller (with a different name).

Relevant to the previous comment, will be introduced as part of that commit.

Reply:
> I agree, I wasn't sure whether to introduce it during this PR or not but we can do that.

The direction sounds good to me.

> My idea was for a new package under virt-handler (e.g. NodeCapabilityManager) and slowly migrate cpu_plugin.go logic into it WDYT?

A general note about future plans: we can either review this PR individually or we need to discuss the overall plan and review your PRs with more context. As suggested in other places, I think a design proposal would be beneficial in this case, as it will allow you to work with smaller PRs and ease the review process.

Reply:
As I've responded in the tracker issue https://github.com/kubevirt/kubevirt/issues/12727, I am currently working on a design proposal to better discuss the overhaul in its entirety. 

For the time being, I believe this PR, even in its standalone form, is essential for any future changes pertaining to the node-labeller.

Reply:
> As I've responded in the tracker issue https://github.com/kubevirt/kubevirt/issues/12727, I am currently working on a design proposal to better discuss the overhaul in its entirety.

Happy to hear that!
Sorry for missing your response.

> For the time being, I believe this PR, even in its standalone form, is essential for any future changes pertaining to the node-labeller.

I agree, introducing the design proposal shouldn't block this PR.
Thanks!